### PR TITLE
Remove t prefix

### PR DESCRIPTION
**Pull request by Perelandric**
Update library to drop `T` from types. #2773  Everything builds on Linux. Haven't built on Windows.

Wasn't able to change `TGenericSeq`, `TNimType`, `TSafePaint` or `TFrame` successfully.

Some files contain types that cause conflict if the `T` is dropped:

 - posix/posix.nim
 - wrappers/expat.nim
 - wrappers/libuv.nim
 - wrappers/odbcsql.nim
 - wrappers/zzip.nim
 - windows/windows.nim

Search for the word "conflict" in each file to find the conflicting types.

In some files (especially windows/windows.nim), some `TFoo` types had been fixed as follows:
```
type
  Foo* = object

TFoo* = Foo
```

In those cases, I removed the `TFoo* = Foo` and replaced them with `{.deprecated: [TFoo: Foo].}`.

### CODE REVIEW

**Comment by yglukhov** [comment]
May I propose not touching js/dom.nim for now? The types there should be changed to have reference semantics.

File: lib/impure/zipfiles.nim [hunk 1]
```diff
@@ -109,12 +109,13 @@ proc addFile*(z: var TZipArchive, dest: string, src: Stream) =
...
 type
-  TZipFileStream = object of StreamObj
+  ZipFileStream = object of StreamObj
     f: PZipFile
     atEnd: bool
 
   PZipFileStream* =
-    ref TZipFileStream ## a reader stream of a file within a zip archive
+    ref ZipFileStream ## a reader stream of a file within a zip archive
+{.deprecated: [TZipFileStream: ZipFileStream].}
```
Comment (Araq) [inline lib/impure/zipfiles.nim]: No, ZipFileStream should be the ref type.

File: lib/js/dom.nim [hunk 2]
```diff
@@ -449,14 +449,20 @@ type
     pixelDepth*: int
     width*: int
 
-  TTimeOut* {.importc.} = object of RootObj
-  TInterval* {.importc.} = object of RootObj
+  TimeOut* {.importc.} = object of RootObj
+  Interval* {.importc.} = object of RootObj
+{.deprecated: [TEventHandlers: EventHandlers, TEvent: Event, TNodeType: NodeType,
```
Comment (Araq) [inline lib/js/dom.nim]: I think the names without prefixes should refer to the 'ref' types for JS.

File: lib/pure/collections/intsets.nim [hunk 3]
```diff
@@ -30,25 +30,25 @@ const
...
-  TTrunkSeq = seq[PTrunk]
+  TrunkSeq = seq[PTrunk]
   IntSet* = object ## an efficient set of 'int' implemented as a sparse bit set
     counter, max: int
     head: PTrunk
-    data: TTrunkSeq
+    data: TrunkSeq
 
-{.deprecated: [TIntSet: IntSet].}
+{.deprecated: [TIntSet: IntSet, TTrunk: Trunk, TTrunkSeq: TrunkSeq].}
```
Comment (Araq) [inline lib/pure/collections/intsets.nim]: These were never exported in the first place.

File: lib/pure/strutils.nim [hunk 4]
```diff
@@ -23,7 +23,8 @@ import parseutils
 include "system/inclrtl"
 
 type
-  TCharSet* {.deprecated.} = set[char] # for compatibility with Nim
+  CharSet* {.deprecated.} = set[char] # for compatibility with Nim
+{.deprecated: [TCharSet: CharSet].}
```
Comment (Araq) [inline lib/pure/strutils.nim]: Just remove it instead.

File: lib/system/excpt.nim [hunk 5]
```diff
@@ -365,5 +365,6 @@ when not defined(noSignalHandler):
 
 proc setControlCHook(hook: proc () {.noconv.} not nil) =
   # ugly cast, but should work on all architectures:
-  type TSignalHandler = proc (sig: cint) {.noconv, benign.}
-  c_signal(SIGINT, cast[TSignalHandler](hook))
+  type SignalHandler = proc (sig: cint) {.noconv, benign.}
+  {.deprecated: [TSignalHandler: SignalHandler].}
```
Comment (Araq) [inline lib/system/excpt.nim]: Pointless .deprecated.

**Comment by Perelandric** [comment]
@Araq - I was only concerned with renaming. Are you sure you want to mix semantic changes in with this PR? Wouldn't a clean separation be desirable?

I wasn't inclined to remove a previously deprecated type that needed a rename. Seems arbitrary to do it here.

Also, there are some pointless changes, like places where unexported types were deprecated. I figured a deprecated warning would be preferred if I missed a place that used the old type or if someone in the future added the old one.

That said... I'll make your noted changes if you'd like.

**Comment by yglukhov** [comment]
@PavelVozenilek, types that are subject to semantic changes should not be touched in this PR, so that their future desired names (with adjusted semantics) are not reserved.

**Comment by kirbyfan64** [comment]
@yglukhov I think you pinged the wrong person.

**Comment by Araq** [comment]
@Perelandric what @yglukhov said.

**Comment by Araq** [comment]
@dom96  @Varriount  please review too, looks good to me, but it's a huge PR.

**Comment by dom96** [comment]
I think it looks good.

**Comment by Araq** [comment]
Merged manually.

**Pull request closed without merging**